### PR TITLE
feat: Update Conversations SDK to 4.0.3

### PR DESCRIPTION
**Pull request by AlejandroOrozco**
**Context:**
Since the latest version the `type` (ChannelType) property from the Participant is no longer available from Swift context, instead it was replaced for a basic enum for Swift. Previously the `type` was being stored in CoreData as an integer, so the idea behind this update is to stored the channel as a plain String, since it will be use only for display purposes.

**Changes:**
- Upgrade Conversations SDK to 4.0.3
- Add data migration to change `type` for the String version.
- Store channel type.

| Type Marked as Unavailable |  StoredAsInt16 |  Channel Type BasicEnum |
|:--:|:--:|:--:|
|Marked as Unavaible for Swift|StoredAsInt16|CreatedAsBasicEnum|

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by boy1bot2** [comment]
https://patch-diff.githubusercontent.com/raw/twilio/twilio-conversations-demo-ios-swift/pull/22.patch

File: ConversationsApp/Models/CoreData/ChannelTypeDataMigration.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        let channel: String?
+
+        switch Int(previousChannelType) {
+            case 0:
+                channel = nil
+            case 1:
+                channel = "other"
+            case 2:
+                channel = "chat"
+            case 3:
```
Comment (rusmonster) [inline ConversationsApp/Models/CoreData/ChannelTypeDataMigration.swift:34]: As I see from [this screenshot](https://private-user-images.githubusercontent.com/4898461/286286439-3428daa0-5ebe-47af-8bdf-31db65c59949.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODkxOTAsIm5iZiI6MTcxNzU4ODg5MCwicGF0aCI6Ii80ODk4NDYxLzI4NjI4NjQzOS0zNDI4ZGFhMC01ZWJlLTQ3YWYtOGJkZi0zMWRiNjVjNTk5NDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDVUMTIwMTMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDE0MGRiMjVlZWQ2NWNjODk2ZmUzZTM4YWY1OTdkY2I2OWE0OWRhZTU4N2JhYmZhNjc5OTFkNjQ4NjgxMDEzYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IhM_OB0qFUg6jhBNM01iWwHrvvheIY0l-GYSSDD_ifA) is should be like:
```
case 3:
   channel = "sms"
case 4:
   channel = "whatsapp"
```
isn't it?

File: ConversationsApp/Models/CoreData/PersistentParticipantDataItem+CoreDataExt.swift [hunk 2]
```diff
@@ -34,8 +34,9 @@ extension PersistentParticipantDataItem {
             self.sid = participant.sid
             self.conversationSid = conversationSid
             self.identity = participant.identity
-            self.type = Int16(participant.type.rawValue)
-            
+            // TODO: Properly convert this channelType to a string, currently it is just a Swift enum and the channel string is hidden for Swift.
+            self.channel = String(describing: participant.channelType)
```
Comment (rusmonster) [inline ConversationsApp/Models/CoreData/PersistentParticipantDataItem+CoreDataExt.swift:38]: Why don't use just [Participant.channel](https://sdk.twilio.com/ios/conversations/releases/4.0.3/docs/Classes/TCHParticipant.html#//api/name/channel) here?

**Review by rusmonster — CHANGES_REQUESTED** [review]

File: ConversationsApp/Models/CoreData/PersistentParticipantDataItem+CoreDataExt.swift (hunk 2 above)
Comment (AlejandroOrozco) [reply ConversationsApp/Models/CoreData/PersistentParticipantDataItem+CoreDataExt.swift:38]: That variable is not visible in the Swift Context, I think the target is missing the bridging-header to automate that translation between Objc & Swift.

Screenshot 2024-06-13 at 5 57 42 PM

File: ConversationsApp/Models/CoreData/ChannelTypeDataMigration.swift (hunk 1 above)
Comment (AlejandroOrozco) [reply ConversationsApp/Models/CoreData/ChannelTypeDataMigration.swift:34]: I think you're right, added both. Thank you 👍

**Review by rusmonster — APPROVED** [review]

**Pull request merged**